examples/postgres/custom_arrays/README.md

```diff
@@ -0,0 +1,1162 @@
+    pub mod sql_types {
+        #[derive(diesel::query_builder::QueryId, Clone, diesel::sql_types::SqlType)]
+        #[diesel(postgres_type(name = "service_endpoint", schema = "smdb"))]
+        pub struct ServiceEndpoint;
+    }
+
+    diesel::table! {
+        use diesel::sql_types::*;
+        use super::sql_types::ServiceEndpoint;
+
+        smdb.service (service_id) {
+            service_id -> Int4,
+            name -> Text,
+            version -> Int4,
+            online -> Bool,
+            description -> Text,
+            health_check_uri -> Text,
+            base_uri -> Text,
+            dependencies -> Array<Nullable<Int4>>,
+            endpoints -> Array<Nullable<ServiceEndpoint>>,
+        }
+    }
+}
+```
+
+Notice, the Postgres types are stored in Postgres therefore you are not seeing them in the generated schema. Only tables
+will show up in the generates schema. Furthermore, you will need a wrapper struct to map from Rust to the Postgres type.
+For the ServiceEndpoint, Diesel already generated a matching wrapper struct with the correct annotations. The service
+table then uses that wrapper struct “ServiceEndpoint” instead of the native Postgres type to reference the custom type
+in the endpoints array.
+
+You want to attach the generated schema.rs to your lib file to access it from within your crate and stop your IDE from
+showing related warnings
+
+Next, you want to create a folder model with a mod file in it attached to your lib file. This is your place to store all
+Rust type implementations matching all the generated Postgres types and tables.
+
+Lastly, you also want to add a type alias for a postgres database connection in the lib file because that gives you the
+freedom to swap between a normal single connection or a connection pool without updating your type implementation or
+tests.
+The connection pool type alias has ben uncommented, but the type signature of the pooled connection makes it obvious why
+a type alias is a good idea.
+
+At this point, your lib file looks as shown below:
+
+```rust
+mod schema;
+pub mod model;
+
+// Alias for a pooled connection.
+// pub type Connection = diesel::r2d2::PooledConnection<diesel::r2d2::ConnectionManager<diesel::pg::PgConnection>>;
+
+// Alias for a normal, single, connection.
+pub type Connection = PgConnection;
+
+```
+
+## Rust Types
+
+In total, you implement three Rust types:
+
+* Enum: ProtocolType
+* Struct: Endpoint
+* Struct: Service
+
+### Rust Enum: ProtocolType
+
+Diesel needs a wrapper struct to store a custom Enum in Postgres, so let’s add one:
+
+```rust
+#[derive(SqlType)]
+#[diesel(sql_type = protocol_type)]
+#[diesel(postgres_type(name = "protocol_type", schema = "smdb"))]
+pub struct PgProtocolType;
+```
+
+It is important that you add both, sql_type and postgres_type, otherwise insert will fail.
+Furthermore, it is important to add the database schema ("smdb") otherwise Postgres fails to find the type and aborts an
+operation on that type.
```

Comment:
```suggestion
It is important to add the database schema ("smdb") and type name ("protocol_type") otherwise Postgres fails to find the type and aborts an
operation on that type.
```